src/main/java/fr/insee/genesis/configuration/auth/security/OIDCSecurityConfig.java

```diff
@@ -1,48 +1,120 @@
 import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
 import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationConverter;
 import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
 import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
 
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
 @Configuration
 @EnableWebSecurity
+@EnableMethodSecurity(prePostEnabled = true)
 @Slf4j
 @ConditionalOnProperty(name = "fr.insee.genesis.authentication", havingValue = "OIDC")
+@ConfigurationProperties(prefix = "fr.insee.genesis.security")
+@RequiredArgsConstructor
 public class OIDCSecurityConfig {
 
-    Config config;
-    @Autowired
-    public OIDCSecurityConfig(Config config) {
-        this.config = config;
-    }
+    @Getter
+    @Setter
+    private String[] whitelistMatchers;
+    private static final String ROLE_PREFIX = "ROLE_";
+    private final RoleConfiguration roleConfiguration;
+    private final SecurityTokenProperties inseeSecurityTokenProperties;
 
-        @Bean
-        public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
-            http
-                    .csrf(AbstractHttpConfigurer::disable)
-                    .sessionManagement(sess -> sess.sessionCreationPolicy(SessionCreationPolicy.STATELESS));
-            for (var pattern : config.getWhiteList()) {
-                http.authorizeHttpRequests(authorize ->
-                        authorize
-                                .requestMatchers(AntPathRequestMatcher.antMatcher(pattern)).permitAll()
-                );
-            }
-            http
-                    .authorizeHttpRequests(configurer -> configurer
-                            .anyRequest().authenticated()
-                    )
-                    .oauth2ResourceServer(oauth2 -> oauth2.jwt(Customizer.withDefaults()));
-            return http.build();
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+        http
+                .csrf(AbstractHttpConfigurer::disable)
+                .sessionManagement(sess -> sess.sessionCreationPolicy(SessionCreationPolicy.STATELESS));
+        for (var pattern : whitelistMatchers) {
+            http.authorizeHttpRequests(authorize ->
+                    authorize
+                            .requestMatchers(AntPathRequestMatcher.antMatcher(pattern)).permitAll()
+            );
         }
+        http
+                .authorizeHttpRequests(configurer -> configurer
+                        .requestMatchers("/questionnaires/**").hasRole(String.valueOf(ApplicationRole.READER))
```

Comment:
.requestMatchers(HttpMethod.GET, "/questionnaires/**") <= to add GET method only